File: lib/data/repositories/alarm_repository.dart
import '../datasources/local_datasource.dart';
import '../models/alarm_model.dart';

class AlarmRepository {
  final LocalDataSource _localDataSource;

  AlarmRepository(this._localDataSource);

  Future<void> addAlarm(AlarmModel alarm) {
    return _localDataSource.addAlarm(alarm);
  }

  Future<AlarmModel?> getAlarm(String id) {
    return _localDataSource.getAlarm(id);
  }

  List<AlarmModel> getAllAlarms() {
    return _localDataSource.getAllAlarms();
  }

  Future<void> updateAlarm(AlarmModel alarm) {
    return _localDataSource.updateAlarm(alarm);
  }

  Future<void> deleteAlarm(String id) {
    return _localDataSource.deleteAlarm(id);
  }
} 

File: app/src/main/java/com/example/geo_alarm/data/AlarmRepository.kt
package com.example.geo_alarm.data

import kotlinx.coroutines.flow.Flow

class AlarmRepository(private val alarmDao: AlarmDao) {
    val allAlarms: Flow<List<Alarm>> = alarmDao.getAllAlarms()

    suspend fun getAlarm(id: String): Alarm? {
        return alarmDao.getAlarmById(id)
    }

    suspend fun insert(alarm: Alarm) {
        alarmDao.insertAlarm(alarm)
    }

    suspend fun delete(alarm: Alarm) {
        alarmDao.deleteAlarm(alarm)
    }

    suspend fun update(alarm: Alarm) {
        alarmDao.updateAlarm(alarm)
    }
}
